### PR TITLE
First api endpoints implemented in the backend

### PR DESCRIPTION
**Pull request by Gbury**
This PR implements the first few API endpoints: listing all events, querying the details (for now just the name) of an event, and creating a new event.

With this PR, the openapi doc (in json format) is now automatically generated by the backend, and can be accessible at `localhost:8080:openapi.json` when running the server (using the `make test` target).

### CODE REVIEW

File: src/backend/event.ml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+(* Event creation *)
+(* ************************************************************************* *)
+
+and create_event =
+  Api.put
+    ~of_yojson:Types.Event.of_yojson
+    ~to_yojson:Types.EventId.to_yojson
+    (fun _req st (event : Types.Event.t) ->
+       let start_date = Ftw.Date.mk ~day:1 ~month:1 ~year:2025 in
+       let end_date = Ftw.Date.mk ~day:1 ~month:1 ~year:2025 in
```
Comment (MagicalLay) [inline src/backend/event.ml:109]: Do we really want to be able to create an event without providing start and end dates? It seems a bit weird to me

File: src/backend/types.ml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        "events", obj @@ S.make_schema ()
+          ~typ:array
+          ~items:(ref EventId.ref);
+      ]
+end
+
+(* Event specification *)
+module Event = struct
+  type t = {
+    name : string;
```
Comment (MagicalLay) [inline src/backend/types.ml:112]: shouldn't you add start_date and end_date there as well ? to comply with api endpoints declared above

File: src/backend/event.ml (hunk 1 above)
Comment (Gbury) [reply src/backend/event.ml:109]: I'm adding that right now

**Review by MagicalLay — APPROVED** [review]
apart from the points mentioned in individual comment, it looks good to me. You talk about those dates in the PR description, but it is the only inconsistent point inside the PR itself (used sometimes but not always)

File: src/backend/event.ml (hunk 1 above)
Comment (Gbury) [reply src/backend/event.ml]: fixed by 8cd1695

File: src/backend/types.ml (hunk 2 above)
Comment (Gbury) [reply src/backend/types.ml:137]: done

**Comment by Gbury** [comment]
The last commit adds dates to the event creation and query, ^^

**Pull request merged**